app/src/main/java/nodomain/freeyourgadget/gadgetbridge/service/devices/hplus/HPlusSupport.java

```diff
@@ -882,11 +889,33 @@ private void  processExtraInfo (byte[] data) {
         }
     }
 
-    private void handleBatteryInfo(byte data) {
-            if (batteryCmd.level != (short) data) {
-                batteryCmd.level = (short) data;
-                handleGBDeviceEvent(batteryCmd);
+    private String getLastCharge(byte data) {
```

Comment:
The method name implies a simple conversion/calculation method. Could you rename it so that one becomes aware of its side effects (e.g. `determineLastChargeTimeAndNotify()`). A bit long, yes, but that's what you get when doing multiple things in a single method :-]